tests/chainer_tests/test_variable.py

```diff
@@ -2742,26 +2742,47 @@ def test_grad_raise_double_backprop_2(self):
             chainer.grad([x.grad_var], [y.grad_var])
 
 
+@testing.backend.inject_backend_tests(None, _backend_params)
 class TestAsVariable(unittest.TestCase):
 
-    def check_to_variable_from_array(self, x):
+    def test_to_variable_from_array(self, backend_config):
+        x = backend_config.get_array(np.empty(1, np.float32))
         y = chainer.as_variable(x)
         assert isinstance(y, chainer.Variable)
-        assert y.data is x
-        assert not y.requires_grad
+        assert y.requires_grad is False
+
+        if backend_config.xp is chainerx:
+            # chainerx
+            assert y.array.shape == x.shape
+            assert y.array.device == x.device
+            assert y.array.strides == x.strides
+            assert not y.array.is_backprop_required()
+            chainerx.testing.assert_array_equal(y.array, x)
```

Comment:
I couldn't find the doc of `chainerx.testing`.  Is it guaranteed that NaNs are equal (as in `numpy.testing.assert_array_equal`)?

Reply:
Yes, it behaves just as `numpy.testing.assert_array_equal`, except inputs are `chainerx.ndarray`s.
But maybe it's better to avoid `empty`.

Reply:
Done.